### PR TITLE
Home page touch-ups

### LINKED ISSUE #2133: Save and load a snapshot of last-opened part in each project to display on the home page
[Figma mockup](https://www.figma.com/file/5UUPOawPLEs3D0Gptf42vy/Designs?type=design&node-id=1720-82784&mode=design&t=5tJtGQA3qIlVkpYX-11)

Original note from @JBEmbedded:
> Preview image of project on each project card as default, with icons to be able to choose whether you want projects displayed as a grid or as list. Image preview hidden in list view.

### PR DESCRIPTION
**Pull request by franknoirot**
Start of an implementation toward #2133. Dialed back this to not include the screenshot functionality so we could land the PR sooner rather than later. ~~However, I'm going to keep this in draft form until we have a "zoom to fit" endpoint in the engine, and then use the [`take_snapshot`](https://zoo.dev/docs/api/modeling/open-a-websocket-which-accepts-modeling-commands?lang=python#parameters-body-modeling_cmd_req-cmd-take_snapshot) modeling command to get a much more consistent snapshot.~~

~~Another design dilemma I'm in is how to capture new snapshots from the home page. I want to make this more robust, so that when either:~~
1. ~~the `app.theme` setting is changed~~
2. ~~a project does not yet have a thumbnail
the card is put into a loading state and a new image is captured without leaving the home page~~

## Demo with current limitations

https://github.com/KittyCAD/modeling-app/assets/23481541/335d1401-186f-4d42-ae48-1f0f4e8a3f2a

### CODE REVIEW

**Comment by franknoirot** [comment]
Some open questions:
1. How do we take snapshots from the home page?
2. Is there a different method other than saving a frame from the stream that allows me to have a transparent background to these images?
3. Do we want users to be able to self-author their file thumbnails via a menu?
  1. How would that self-authorship work with default thumbnail capture behavior?

**Comment by Irev-Dev** [comment]
we can definitely store the camera position, which will go more neatly into the project.toml than an actually image would.

And `take_snapshot` is the endpoint used for taking a snapshot (websocket endpoint), pretty sure that's the same thing the cli uses, so you could try that instead when the user leaves to get something transparent. Updating all of the snapshots everytime the user changes the theme sounds messy to me.

Getting snap shots for projects the user hasn't open 🤷 not sure hey.

**Comment by Irev-Dev** [comment]
image

image

**Comment by franknoirot** [comment]
> we can definitely store the camera position, which will go more neatly into the project.toml than an actually image would.

Sorry I used imprecise language in the video, I meant just save the path to the image in `project.toml`. But that should just be a constant anyway.

I like the idea of using `take_snapshot` to take a PNG, and setting the camera's position for the project thumbnail in `project.toml`. If they haven't manually `Set project thumbnail` we have some sensible camera default camera position, and regardless we always call `zoom_to_fit` to get the whole geometry in view once that's available.

**Comment by franknoirot** [comment]
This PR resolves #2337

File: src/hooks/useSaveVideoFrame.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import { RefObject, useEffect } from 'react'
```
Comment (jessfraz) [inline src/hooks/useSaveVideoFrame.ts:1]: if we arent doing the save here can we remove this, just want to not inject bugs, not saying its buggy just would rather it be in once we get the current feature set working well

Comment (franknoirot) [reply src/hooks/useSaveVideoFrame.ts:1]: Yeah totally I'll just drop it!

**Pull request merged**